e2e/util_test.go

```diff
@@ -0,0 +1,109 @@
+package e2e
+
+import (
+	"context"
+	"crypto/rand"
+	_ "embed"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/authzed/authzed-go/v1"
+	"github.com/authzed/grpcutil"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/ory/dockertest/v3"
+	"github.com/rs/zerolog/log"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+//go:embed fixtures/test_schema.sql
+var testSchema string
+
+func spicedb(require *require.Assertions) (*authzed.Client, func()) {
+	pool, err := dockertest.NewPool("")
+	require.NoError(err)
+
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Cmd:          strings.Split(`serve --grpc-preshared-key somerandomkeyhere --grpc-no-tls --http-no-tls`, " "),
+		Repository:   "quay.io/authzed/spicedb",
+		Tag:          "v1.1.0",
+		ExposedPorts: []string{"50051"},
+	})
+	require.NoError(err)
+
+	var client *authzed.Client
+	port := resource.GetPort("50051/tcp")
+	require.NoError(pool.Retry(func() error {
+		var err error
+		client, err = authzed.NewClient(fmt.Sprintf("localhost:%s", port), grpcutil.WithInsecureBearerToken("somerandomkeyhere"), grpc.WithInsecure())
+		if err != nil {
+			log.Debug().Err(err).Send()
+			return err
+		}
+		return nil
+	}))
+
+	cleanup := func() {
+		require.NoError(pool.Purge(resource))
+	}
+	return client, cleanup
```

Comment:
if we did t.Cleanup we might not even have to return this closure

Reply:
t.Cleanup makes this much nicer